js/queries.js

```diff
@@ -51,14 +58,18 @@ function getSourceText(state, actor) {
 }
 
 function getBreakpoint(state, location) {
+  location = Immutable.fromJS(location);
   return getBreakpoints(state).get(makeLocationId(location));
 }
 
 /**
  * @param object - location
  */
 function makeLocationId(location) {
-  return location.actor + ":" + location.line.toString();
+  if(!Immutable.Iterable.isIterable(location)) {
+    location = Immutable.fromJS(location);
+  }
+  return location.get("actor") + ":" + location.get("line").toString();
```

Comment:
There is still some mismatch in a few places. This is what I was referencing in the first paragraph of this PR description. We should not need to coerce here; it was a stop-gap. There seemed to be one or two places where a non-immutable value would come through here but it wasn't straight-forward how.

I'll look at this more now and see if I can't do a more thorough conversion to immutable.

We need to decide which values are immutable or not. It's clear that if our state is immutable, components will receive immutable values and use them directly. But I think any parameters to actions or queries should be normal JS objects because you usually pass literals in (`action({ line: 5})`).